node/runtime/src/impls.rs

```diff
@@ -130,81 +163,120 @@ mod tests {
 		Perbill::from_parts((fm * 1_000_000_000_f32) as u32)
 	}
 
-	fn fm(parts: i64) -> WeightMultiplier {
+	fn wm(parts: i64) -> WeightMultiplier {
 		WeightMultiplier::from_parts(parts)
 	}
 
+	#[test]
+	fn empty_chain_simulation() {
+		// just a few txs per_block.
+		let block_weight = 1000;
+		let mut wm = WeightMultiplier::default();
+		let mut iterations: u64 = 0;
+		loop {
+			let next = WeightMultiplierUpdateHandler::convert((block_weight, wm));
+			wm = next;
+			if wm == WeightMultiplier::from_rational(-1, 1) { break; }
+			iterations += 1;
+		}
+		println!("iteration {}, new wm = {:?}. Weight fee is now zero", iterations, wm);
```

Comment:
yeah, that's the purpose. And I am adding more. These are not benchmarks, but rather sort of a report. I couldn't think of putting an assertion into it since it is really not checking anything but definitely helpful to have them as reports. 

Though I can add assertions but then it would be a bit based on hardcoded values of `substrate-node` + it will be a damn slow test.

For now I am happy to have them as `ignore` tests that one can run manually to just check things.

Reply:
@Kianenigma what about using `debug!` from the `log` crate?
```suggestion
		debug!("iteration {}, new wm = {:?}. Weight fee is now zero", iterations, wm);
```

Reply:
I don't really see a reason for it tbh, Logging is great when you want sorting, structure, levels, filtering, or if you have a very high volume etc. This is neither of them. In some tests, something is printed _literally just once_. I have no problem in changing it but I think  it is a bit of an overdue.

Reply:
agree. logging in tests is a royal pain since you have to link the log module and research that magical incantation to put at the beginning of the test in order to make the log do anything at all. `println`s are fine.